### ISSUE TITLE
pre-trained meta models

### ISSUE THREAD

**Issue by zhangmaoxiansheng** [sanowar-raihan/nerf-meta#1]
HI. Thanks for your pytorch code. Can you share the pre-trained meta models? Besides, I found the training is not very stable, sometimes the val PSNR keeps the same number while training and the test results are just some lumps.

**Comment by munshisanowarraihan** [comment]
Hi, thank you for your interest!

1. For most of my experiments, I used a lighter model (4 hidden layers, trained for half as many epochs). Even the results in the readme as generated with this lightweight model. So, I don't have the weight files for the full model mentioned in the config. But feel free to contribute the weights if you can manage to train the full model.

2. Instability in NeRF training has been observed before. For example, have a look at the issues [here](https://github.com/kwea123/nerf_pl/issues/46), [here](https://github.com/kwea123/nerf_pl/issues/75) and [here](https://github.com/bmild/nerf/issues/29). This is largely because of using ReLU activation for sigma output. If you have an "unlucky" initialization, sigma will be zero at the beginning and training will not converge.  
You can replace `F.relu()` in the line below with `F.softplus()` and that should work better:
https://github.com/sanowar-raihan/nerf-meta/blob/85418c06f4868dcedc077749a21ab785065ac92b/models/nerf.py#L75    
[mip-NeRF](https://arxiv.org/abs/2103.13415) paper also discusses this instability issue and encourages the use of Softplus. From section E.2 : 

    > In the original NeRF paper, the activation functions used by the MLP to construct the predicted density τ and color c are a ReLU and a sigmoid, respectively. Instead of a ReLU as the activation function to produce τ , we use a shifted softplus: log(1 + exp(x − 1)). We found that using a softplus yielded a smoother optimization problem that is less prone to catastrophic failure modes in which the MLP emits negative values everywhere (in which case all gradients from τ are zero and optimization will fail).

**Comment by zhangmaoxiansheng** [comment]
Thanks for your advice and it does work. Softplus is magic!
